contrib/ssssss.psgi

```diff
@@ -80,17 +93,53 @@ sub handle_options
     return $response;
 }
 
+sub check_authorization_header
+{
+    my ($request) = @_;
+    my $auth = $request->header('authorization');
+
+    my ($user, $pass) = $auth =~ /^LOW ([^:]+):(.+)$/;
+    if (
+        !defined $user ||
+        !defined $pass ||
+        $user ne DBDefs->COVER_ART_ARCHIVE_ACCESS_KEY ||
+        $pass ne DBDefs->COVER_ART_ARCHIVE_SECRET_KEY
+    ) {
+        my $response = $request->new_response(403);
+        $response->content_type('text/xml');
+        $response->body(q{<?xml version='1.0' encoding='UTF-8'?>
+            <Error><Code>AccessDenied</Code></Error>});
+        return $response;
+    }
+}
+
 sub handle_put
 {
     my ($request) = @_;
 
-    my $bucketdir = create_bucket($request->path_info);
-    my $dest = catfile($bucketdir, $request->param('file'));
-    $log->info("PUT, storing upload at $dest\n");
+    my $response = check_authorization_header($request);
+    if ($response) {
+        return $response;
+    }
 
-    open(my $fh, ">", $dest);
-    print $fh $request->content;
-    close($fh);
+    my $file = $request->param('file');
+    if ($file) {
+        my $bucketdir = create_bucket($request->path_info);
+        my $dest = catfile($bucketdir, $file);
+        $log->info("PUT, storing upload at $dest\n");
+
+        open(my $fh, ">", $dest);
+        print $fh $request->content;
+        close($fh);
+    } elsif (bucket_exists($request->path_info)) {
+        $response = $request->new_response(409);
+        $response->content_type('text/xml');
+        $response->body(q{<?xml version='1.0' encoding='UTF-8'?>
+            <Error><Code>BucketAlreadyExists</Code></Error>});
+        return $response;
+    } else {
+        create_bucket($request->path_info);
+    }
 
     return $request->new_response(204);
 }
```

Comment:
Can we log these errors using Sentry? Or do we have access to them elsewhere already?

Reply:
We'd need to add some methods to MusicBrainz::Errors to allow logging arbitrary messages to Sentry. I don't have the means to test that right now, but I could look into it after the schema change.

(But we wouldn't want to log all `BucketAlreadyExists` errors; only cases where the ownership is wrong. And that should rarely ever happen, hopefully, other than the one case we know about.)